### PR TITLE
ParallelRunner support for RunFromFileSytem

### PR DESCRIPTION
**Pull request by florintamas**
ParallelRunner support for RunFromFileSytem build.
JIRA ticket: https://issues.jenkins-ci.org/browse/JENKINS-52535

### CODE REVIEW

File: HpToolsLauncher/ParallelRunner/ParallelRunnerEnvironmentUtil.cs [hunk 1]
```diff
@@ -426,9 +430,29 @@ public static string GetConfigFilePath(TestInfo testInfo, McConnectionInfo mcCon
...
+            catch(ArgumentException e)
+            {
+                throw new ParallelRunnerConfigurationException("Configuration serialization recursion limit exceeded: ", e);
+            }
+
+            try
+            {
+                File.WriteAllText(configFilePath, configJson);
+            }
+            catch (Exception e)
```
Comment (grount) [inline HpToolsLauncher/ParallelRunner/ParallelRunnerEnvironmentUtil.cs]: Should we classify each exception with other messages for better error handling?

Comment (florintamas) [reply HpToolsLauncher/ParallelRunner/ParallelRunnerEnvironmentUtil.cs]: The inner exception is specified when throwing the ParallelRunnerConfigurationException.

**Comment by florintamas** [comment]
JIRA ticket: https://issues.jenkins-ci.org/browse/JENKINS-52535

**Comment by grount** [comment]
Approved for test build

**Comment by grount** [comment]
@florintamas  You can close the JIRA ticket, and put the PR there when you close it.

**Pull request merged**